karspexet/ticket/views.py

```diff
@@ -11,14 +30,125 @@ def home(request):
         'upcoming_shows': upcoming_shows
     })
 
+
 def select_seats(request, show_id):
     show = Show.objects.get(id=show_id)
-    forms = [SeatingGroupFormSet(seating_group) for seating_group in show.venue.seatinggroup_set.all()]
+    reservation = _get_or_create_reservation_object(request, show)
+    taken_seats_qs = Reservation.active.exclude(pk=reservation.pk).filter(show=show)
+    _set_session_timeout(request)
+
+    if request.POST:
+        seat_params = _seat_specifications(request)
+        if _all_seats_available(taken_seats_qs, seat_params.keys()):
+            reservation.tickets = seat_params
+            reservation.save()
+            return redirect("payment", show_id=show_id)
+        else:
+            messages.error(request, "Some of your chosen seats are already taken")
+
+    taken_seats = set(map(int,set().union(*[r.tickets.keys() for r in taken_seats_qs.all()])))
```

Comment:
I'm thinking I should probably extract this line to at least a separate function. Maybe even a class.